osgp/platform/osgp-adapter-ws-core/src/main/java/org/opensmartgridplatform/adapter/ws/core/endpoints/FirmwareManagementEndpoint.java

```diff
@@ -923,13 +937,24 @@ public ChangeFirmwareResponse changeFirmware(
         this.firmwareManagementMapper.map(
             request.getFirmware().getFirmwareModuleData(), FirmwareModuleData.class);
 
+    // The ChangeFirmwareRequest accepts multiple DeviceModels to be related to a Firmware.
+    // This FirmwareManagementService only accepts ONE for now
+    final List<org.opensmartgridplatform.adapter.ws.schema.core.firmwaremanagement.DeviceModel>
+        wsDeviceModels = request.getFirmware().getDeviceModels();
+    String modelCode = null;
+    String manufacturer = null;
+    if (!wsDeviceModels.isEmpty()) {
+      modelCode = wsDeviceModels.get(0).getModelCode();
+      manufacturer = wsDeviceModels.get(0).getManufacturer();
+    }
```

Comment:
These lines of code are the same as the lines 874 - 884. Consider creating a function.

Reply:
Extracted method for these lines.